Can't build android arm32 template (vulkan error)

### Tested versions

reproduced:
- v4.3.stable.official [77dcf97d8]

### System information

Ubuntu 22.04.4 LTS, GTX1060 6GB, NVIDIA 535.183.01, Compatibility mode

### Issue description

Tried to compile android arm32-template but build stopped with errors.

**command**
```
scons platform=android arch=arm32 target=template_debug generate_apk=yes
```

**output**
```
[ 70%] Compiling scene/gui/item_list.cpp ...
[ 70%] Compiling scene/gui/label.cpp ...
[ 70%] Compiling scene/gui/line_edit.cpp ...
[ 70%] Compiling scene/gui/link_button.cpp ...
drivers/vulkan/rendering_context_driver_vulkan.cpp:240:3: error: no viable conversion from '(lambda at drivers/vulkan/rendering_context_driver_vulkan.cpp:240:3)' to 'PFN_vkAllocationFunction' (aka 'void *(*)(void *, unsigned int, unsigned int, VkSystemAllocationScope) __attribute__((pcs("aapcs-vfp")))')
                [](
                ^~~
drivers/vulkan/rendering_context_driver_vulkan.cpp:240:3: note: candidate function
drivers/vulkan/rendering_context_driver_vulkan.cpp:272:3: error: no viable conversion from '(lambda at drivers/vulkan/rendering_context_driver_vulkan.cpp:272:3)' to 'PFN_vkReallocationFunction' (aka 'void *(*)(void *, void *, unsigned int, unsigned int, VkSystemAllocationScope) __attribute__((pcs("aapcs-vfp")))')
                [](
                ^~~
drivers/vulkan/rendering_context_driver_vulkan.cpp:272:3: note: candidate function
drivers/vulkan/rendering_context_driver_vulkan.cpp:306:3: error: no viable conversion from '(lambda at drivers/vulkan/rendering_context_driver_vulkan.cpp:306:3)' to 'PFN_vkFreeFunction' (aka 'void (*)(void *, void *) __attribute__((pcs("aapcs-vfp")))')
                [](
                ^~~
drivers/vulkan/rendering_context_driver_vulkan.cpp:306:3: note: candidate function
drivers/vulkan/rendering_context_driver_vulkan.cpp:325:3: error: no viable conversion from '(lambda at drivers/vulkan/rendering_context_driver_vulkan.cpp:325:3)' to 'PFN_vkInternalAllocationNotification' (aka 'void (*)(void *, unsigned int, VkInternalAllocationType, VkSystemAllocationScope) __attribute__((pcs("aapcs-vfp")))')
                [](
                ^~~
drivers/vulkan/rendering_context_driver_vulkan.cpp:325:3: note: candidate function
drivers/vulkan/rendering_context_driver_vulkan.cpp:331:3: error: no viable conversion from '(lambda at drivers/vulkan/rendering_context_driver_vulkan.cpp:331:3)' to 'PFN_vkInternalFreeNotification' (aka 'void (*)(void *, unsigned int, VkInternalAllocationType, VkSystemAllocationScope) __attribute__((pcs("aapcs-vfp")))')
                [](
                ^~~
drivers/vulkan/rendering_context_driver_vulkan.cpp:331:3: note: candidate function
5 errors generated.
scons: *** [drivers/vulkan/rendering_context_driver_vulkan.android.template_debug.arm32.o] Error 1
```

however `scons platform=android arch=arm64 target=template_debug generate_apk=yes` compiles without problem

### Steps to reproduce

try to compile with `scons platform=android arch=arm32 target=template_debug generate_apk=yes`

### Minimal reproduction project (MRP)

- n/a

--- a/drivers/vulkan/rendering_context_driver_vulkan.cpp
+++ b/drivers/vulkan/rendering_context_driver_vulkan.cpp
@@ -229,6 +229,13 @@ VkAllocationCallbacks *RenderingContextDriverVulkan::get_allocation_callbacks(Vk
 #if !defined(VK_TRACK_DRIVER_MEMORY)
 	return nullptr;
 #else
+
+#ifdef _MSC_VER
+#define LAMBDA_VK_CALL_CONV
+#else
+#define LAMBDA_VK_CALL_CONV VKAPI_PTR
+#endif
+
 	struct TrackedMemHeader {
 		size_t size;
 		VkSystemAllocationScope allocation_scope;
@@ -241,7 +248,7 @@ VkAllocationCallbacks *RenderingContextDriverVulkan::get_allocation_callbacks(Vk
 				void *p_user_data,
 				size_t size,
 				size_t alignment,
-				VkSystemAllocationScope allocation_scope) -> void * {
+				VkSystemAllocationScope allocation_scope) LAMBDA_VK_CALL_CONV -> void * {
 			static constexpr size_t tracking_data_size = 32;
 			VkTrackedObjectType type = static_cast<VkTrackedObjectType>(*reinterpret_cast<VkTrackedObjectType *>(p_user_data));
 
@@ -274,7 +281,7 @@ VkAllocationCallbacks *RenderingContextDriverVulkan::get_allocation_callbacks(Vk
 				void *p_original,
 				size_t size,
 				size_t alignment,
-				VkSystemAllocationScope allocation_scope) -> void * {
+				VkSystemAllocationScope allocation_scope) LAMBDA_VK_CALL_CONV -> void * {
 			if (p_original == nullptr) {
 				VkObjectType type = static_cast<VkObjectType>(*reinterpret_cast<uint32_t *>(p_user_data));
 				return get_allocation_callbacks(type)->pfnAllocation(p_user_data, size, alignment, allocation_scope);
@@ -305,7 +312,7 @@ VkAllocationCallbacks *RenderingContextDriverVulkan::get_allocation_callbacks(Vk
 		// Free function
 		[](
 				void *p_user_data,
-				void *p_memory) {
+				void *p_memory) LAMBDA_VK_CALL_CONV {
 			if (!p_memory) {
 				return;
 			}
@@ -326,13 +333,13 @@ VkAllocationCallbacks *RenderingContextDriverVulkan::get_allocation_callbacks(Vk
 				void *p_user_data,
 				size_t size,
 				VkInternalAllocationType allocation_type,
-				VkSystemAllocationScope allocation_scope) {
+				VkSystemAllocationScope allocation_scope) LAMBDA_VK_CALL_CONV {
 		},
 		[](
 				void *p_user_data,
 				size_t size,
 				VkInternalAllocationType allocation_type,
-				VkSystemAllocationScope allocation_scope) {
+				VkSystemAllocationScope allocation_scope) LAMBDA_VK_CALL_CONV {
 		},
 	};
 

